### PR TITLE
Write calendly endpoints

### PR DESCRIPTION
**Pull request by calebschoepp**
# Overview

Endpoints to interact with calendly resource. Note that to speed up the implementation I didn't differentiate between my/any endpoints. Nothing particularly confidential about it anyways.

# Changes

- POST endpoint
- GET endpoint
- PATCH endpoint

# Questions & Notes

- Next PR is going to add authorization.
- I skipped testing for speed of implementation
- Dependent on schema migration PR #220

### CODE REVIEW

File: api/db/migrations/003_add_calendly_resource.down.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DROP TRIGGER IF EXISTS update_time_calendlys ON calendlys;
+
+DROP TABLE IF EXISTS calendlys;
```
Comment (cyrusdiego) [inline api/db/migrations/003_add_calendly_resource.down.sql:3]: nit: newline

File: api/src/repositories/calendlyRepository.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if (link !== undefined && link !== null) {
+        colOptions.link = link;
+    }
+    if (interviewer !== undefined && interviewer !== null) {
+        colOptions.interviewer = interviewer;
+    }
+    if (interviewee !== undefined && interviewee !== null) {
+        colOptions.interviewee = interviewee;
+    }
+    console.log(db.update('calendlys', colOptions, where).compile());
```
Comment (cyrusdiego) [inline api/src/repositories/calendlyRepository.ts:59]: is this console log necessary?

**Review by cyrusdiego — APPROVED** [review]
This LGTM! Just left some comments, other than that I think it's good.

**Review by armiantos — CHANGES_REQUESTED** [review]
Blocking because this will change according with changes in #202 as discussed offline

**Review by armiantos — APPROVED** [review]
LGTM! Let's get this shipped (after adding authorization to calendly routes hahahahha) 🚀

I'm guessing we'll need to implement some sort of queue and run the matching program on our server as a job instead of having the clients patch the calendly object to avoid race conditions.

**Comment by calebschoepp** [comment]
> LGTM! Let's get this shipped (after adding authorization to calendly routes hahahahha) 
> 
> I'm guessing we'll need to implement some sort of queue and run the matching program on our server as a job instead of having the clients patch the calendly object to avoid race conditions.

Going to merge now. Auth will come in follow up PR.

Matching will be done as a one time thing by us. We can write some sort of sql/ts script to do the job for us.

**Pull request merged**